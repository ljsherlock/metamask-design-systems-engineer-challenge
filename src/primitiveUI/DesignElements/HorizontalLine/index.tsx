import React, { FunctionComponent } from 'react'

import {
  propertiesObj,
  StyledComponent,
  styleComponentTypes,
} from '../../../utilities/styleUtility'

interface HorizontalLineProps extends styleComponentTypes {}

let properties = propertiesObj()
properties.layout = {}
properties.space = {}
properties.typography = {}
properties.style = {}
properties.state = {}
properties.variants = {}

const HorizontalLine: FunctionComponent<HorizontalLineProps> = ({
  id,
  className,
  children,
  ...props
}) => {
  return (
    <hr
      id={id}
      className={className}
    />
  )
}
export default StyledComponent(HorizontalLine, 'HorizontalLine')