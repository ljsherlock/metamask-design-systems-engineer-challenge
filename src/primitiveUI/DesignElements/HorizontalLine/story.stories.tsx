import React, {Â FunctionComponent } from 'react'
import {
  withKnobs,
  text,
  select,
  boolean,
} from '@storybook/addon-knobs'

import HorizontalLine from '.'

interface Props {
  name: string
  story: any
}

export const HorizontalLineStory: FunctionComponent<Props> = (props) => {
  return (
    <HorizontalLine/>
  )
}

export default {
  title: 'Components/Design Elements/Horizontal Line',
  component: HorizontalLineStory,
}

// @ts-ignore
HorizontalLineStory.story = {
  name: 'default',
  decorators: [ withKnobs ]
}
