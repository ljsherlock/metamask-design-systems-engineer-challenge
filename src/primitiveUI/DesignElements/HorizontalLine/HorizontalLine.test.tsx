import { shallow } from 'enzyme';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import HorizontalLine from '.'

describe('HorizontalLine', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HorizontalLine />);
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});