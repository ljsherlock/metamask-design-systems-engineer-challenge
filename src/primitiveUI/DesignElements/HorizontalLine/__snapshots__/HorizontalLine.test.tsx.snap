// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalLine renders correctly 1`] = `ShallowWrapper {}`;
